crates/ironrdp-cliprdr/src/backend/mod.rs

```diff
@@ -0,0 +1,140 @@
+//! This module provides infrastructure for implementing OS-specific clipboard backend.
+
+use crate::pdu::{
+    ClipboardFormat, ClipboardFormatId, ClipboardGeneralCapabilityFlags, FileContentsRequest, FileContentsResponse,
+    FormatDataRequest, FormatDataResponse, LockDataId,
+};
+
+pub trait ClipboardError: std::error::Error + Send + Sync + 'static {}
+impl<T> ClipboardError for T where T: std::error::Error + Send + Sync + 'static {}
+
+/// Message sent by the OS clipboard backend event loop.
+#[derive(Debug)]
+pub enum ClipboardMessage {
+    /// Sent by clipboard backend when OS clipboard content is changed and ready to be
+    /// delay-rendered when needed by the remote.
+    ///
+    /// Client implementation should initiate copy on `CLIPRDR` SVC when this message is received.
+    SendInitiateCopy(Vec<ClipboardFormat>),
+
+    /// Sent by clipboard backend when format data is ready to be sent to the remote.
+    ///
+    /// Client implementation should send format data to `CLIPRDR` SVC when this message is
+    /// received.
+    SendFormatData(FormatDataResponse<'static>),
+
+    /// Sent by clipboard backend when format data in given format is need to be received from
+    /// the remote.
+    ///
+    /// Client implementation should send initiate paste on `CLIPRDR` SVC when this message is
+    /// received.
+    SendInitiatePaste(ClipboardFormatId),
+
+    /// Failure received from the OS clipboard event loop.
+    ///
+    /// Client implementation should log/display this error.
+    Error(Box<dyn ClipboardError>),
+}
+
+/// Proxy to send messages from the os clipboard backend to the main application event loop
+/// (e.g. winit event loop).
+pub trait ClipboardMessageProxy: std::fmt::Debug + Send + Sync {
+    fn send_clipboard_message(&self, message: ClipboardMessage);
+}
+
+/// OS-specific clipboard backend interface.
+pub trait CliprdrBackend: std::fmt::Debug + Send + Sync + 'static {
+    /// Returns path to local temporary directory where clipboard-transferred files should be
+    /// stored.
+    fn temporary_directory(&self) -> &str;
+
+    /// Return capabilities of the client.
```

Comment:
Nit: for consistency
```suggestion
    /// Returns capabilities of the client.
```